### DISCUSSION TITLE
Delete or reset/clear database

### DISCUSSION THREAD

**Discussion by tochiedev** [isar/isar#74] [Help]
Hi,

Please what's the proper way to delete (or empty) a database? 

I tried accessing the database directory with getApplicationDocumentsDirectory() from the flutter path_provider plugin which gave me - '/data/user/0/com.xxx.xxx/app_flutter/isar'. Is this the correct path?

**Comment by simc** [comment]
@tochiedev If you don't pass a custom directory to `openIsar` this path is the default location where the database is stored.

Please close the database before you delete the directory.

**Reply by frestoinc** [reply]
@leisim would it be possible to have a `clear() ` or `drop() ` function. Isar will autoincrement id if pass as null, and if developed badly, will make the table keep inserting, which in turn query longer time? Also i don't think mobile devs have access to client db if directory is set to default path?

**Comment by MarcoBeyer** [comment]
You can delete a database by calling close(deleteFromDisk: true) on your instance.
[https://pub.dev/documentation/isar/latest/isar/Isar/close.html](https://pub.dev/documentation/isar/latest/isar/Isar/close.html)
